.github/workflows/coverage.yml

```diff
@@ -0,0 +1,36 @@
+name: Code Coverage Workflow
+
+on:
+  schedule:
+    - cron: '0 1 * * 0'
+  pull_request: # Will disable this once I see the workflow working correctly
+  workflow_dispatch:
+
+jobs:
+  code-coverage:
+    runs-on: [self-hosted, X64]
+    container:
+      image: ghcr.io/espressosystems/nix:2.8.1
+      volumes:
+        - github_nix_281:/nix
+    timeout-minutes: 240
+    steps:
+      - name: Configure Git
+        run: |
+          git config --global url."https://ancient123:${{ secrets.ORG_GITHUB_PAT }}@github.com/".insteadOf git://github.com/
+          git config --global url."https://ancient123:${{ secrets.ORG_GITHUB_PAT }}@github.com/".insteadOf ssh://git@github.com/
+
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Generate coverage reports
+        run: |
+          git config --global --add safe.directory "$PWD"
+          nix run "github:NixOS/nix?ref=1849e6a1f64734c488c2b1469249d65ce08cef93" -- develop .#perfShell -c env SEAHORSE_TEST_TXN_HISTORY_TIME_TOLERANCE=10 cargo-llvm-cov llvm-cov --lib --bins --tests --release --workspace --all-features --lcov --output-path lcov.info -- --test-threads=1
```

Comment:
It's not a requirement. But the coverage stuff does require extra dependencies (like `cargo-llvm-cov`) and Nix makes managing them easily. Plus it means our coverage setup is uniform across repos, since the other repos use Nix. And just in general, it seems like a good idea for Seahorse to have some Nix support